### ISSUE TITLE
MacOS: Make file build error.

### ISSUE THREAD

**Issue by airi-14x** [citra-emu/citra#3680]
Hello.

I was trying to build the Citra file but it's failing. I have updated the GCC to 7. 
I was wondering if it's because I don't have Xcode install on my laptop? 
I tried to get the Xcode app from the App Store but I'm on 10.11.6 so, it doesn't let me unless I update the OS.(Did that once, got on High Sierra and my laptop's applications were incompatible so I reverted back)

I posted the error that I am receiving. Any help is greatly appreciated.
-----------

[ 71%] Building CXX object src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_stream_buffer.cpp.o
In file included from /Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:11:
/Users/Airi/Documents/citra/externals/./nihstro/include/nihstro/shader_bytecode.h:89:5: warning: 
      control may reach end of non-void function [-Wreturn-type]
    }
    ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:128:15: error: 
      use of undeclared identifier 'iter'
        auto [iter, inserted] =
              ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:128:31: error: 
      expected unqualified-id
        auto [iter, inserted] =
                              ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:130:35: error: 
      use of undeclared identifier 'iter'
        ExitMethod& exit_method = iter->second;
                                  ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:131:14: error: 
      use of undeclared identifier 'inserted'
        if (!inserted)
             ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_decompiler.cpp:136:21: warning: 
      36 enumeration values not handled in switch: 'ADD', 'DP3', 'DP4'...
      [-Wswitch]
            switch (instr.opcode.Value()) {
                    ^
2 warnings and 4 errors generated.
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_shader_decompiler.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_manager.cpp:8:
In file included from /Users/Airi/Documents/citra/src/./video_core/renderer_opengl/gl_shader_manager.h:10:
In file included from /Users/Airi/Documents/citra/src/./video_core/renderer_opengl/gl_shader_gen.h:14:
In file included from /Users/Airi/Documents/citra/src/./video_core/shader/shader.h:11:
/Users/Airi/Documents/citra/externals/./nihstro/include/nihstro/shader_bytecode.h:89:5: warning: 
      control may reach end of non-void function [-Wreturn-type]
    }
    ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_manager.cpp:118:15: error: 
      use of undeclared identifier 'iter'
        auto [iter, new_shader] = shaders.emplace(config, OGLShaderStage...
              ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_manager.cpp:118:33: error: 
      expected unqualified-id
        auto [iter, new_shader] = shaders.emplace(config, OGLShaderStage...
                                ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_manager.cpp:119:41: error: 
      use of undeclared identifier 'iter'
        OGLShaderStage& cached_shader = iter->second;
                                        ^
/Users/Airi/Documents/citra/src/video_core/renderer_opengl/gl_shader_manager.cpp:120:13: error: 
      use of undeclared identifier 'new_shader'
        if (new_shader) {
            ^
1 warning and 4 errors generated.
make[2]: *** [src/video_core/CMakeFiles/video_core.dir/renderer_opengl/gl_shader_manager.cpp.o] Error 1
make[1]: *** [src/video_core/CMakeFiles/video_core.dir/all] Error 2
make: *** [all] Error 2

**Comment by mailwl** [comment]
you may dl any xcode version from https://developer.apple.com/download/more/

**Comment by airi-14x** [comment]
Thank you for the link. It would seem that the download is available only to those who are part of Apple's Developer Program. Which I am not.

I was trying to download Xcode 8.2 and its command line support, but any link from there gives 502: Bad Gateway. I tried searching online for the dmg of them but no luck.

According to StackOverflow, those links worked in the past but now, it can be only downloaded by those who are part of the program.

**Comment by mailwl** [comment]
really 502 bad gateway, even on Enterprise account. it's apple site broken, not related with apple programs(xcode is free software). sorry for advice without testing if it works

**Comment by daniellimws** [comment]
This shouldn't be a problem with xcode. Could you paste the output of running cmake to provide more information?

**Comment by airi-14x** [comment]
Ah, I just realised that the prebuild version on the website works. 

@mailwl : It's fine. Thank you for the suggestion. I didn't realise that there was a developer Xcode download available. How odd, I thought it was because I wasn't part of the Developer program haha. 

@daniellimws : It was indeed a problem with the make file. 
`cmake .. -DCMAKE_OSX_ARCHITECTURES="x86_64;x86_64h" -DCMAKE_BUILD_TYPE=Release`
>> This line was giving an error stating that
`/path/to/citra/externals/<some_name_here> does not contain a CMakeLists.txt file.`

I checked the Wiki of this Github and ran the two commands to get the submodule, but it still showed that error.

Thank you for your assistance. I'll close this issue.
